### PR TITLE
Deployment to production

### PR DESCRIPTION
**Pull request by RobertGHippo**
# Description

Deploy IaC/Terraform to production.

## Ticket number

EYQB-345

## Changes

- New job in pipeline, `deploy to production` (copy of `deploy to staging`)
- Removed `as_service_principal_object_id` variable (principal identity can be read directly from the web app)
- Ignore changes to tags set automatically by policy: "Environment", "Product", "Service Offering"
- A couple of instances of ignoring changes to tenant_id (it's very unlikely that it'll change between two automated deployments)
- Removed some key vault access policies that were duplicates that got in each other's way
- Removed deprecated `send_to_subscription_administrator` and `send_to_subscription_co_administrator` notifications

# How Has This Been Tested?

Deployed both dev and staging from pipeline; a couple of outstanding tasks before it can be deployed fully to production (DNS records, certificate generation), but no need to wait for those.

# Checklist:

- [x] My code follows the standards used within this project
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests (Unit, E2E) that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/main-build-and-deploy.yml [hunk 1]
```diff
@@ -120,3 +120,45 @@ jobs:
       - name: Swap to the production slot
         run: |
           az webapp deployment slot swap --resource-group ${{ vars.RESOURCE_NAME_PREFIX }}-rg --name ${{ vars.WEBAPP_NAME }} --slot ${{ vars.WEBAPP_SLOT_NAME }} --target-slot production
+
+  deploy-to-production:
+    runs-on: ubuntu-22.04
+    needs: [create-and-publish-image, deploy-to-dev]
+    name: Deploy to staging
```
Comment (sam-c-dfe) [inline .github/workflows/main-build-and-deploy.yml:127]: Name should be 'Deploy to production'

File: .github/workflows/main-build-and-deploy.yml [hunk 2]
```diff
@@ -120,3 +120,45 @@ jobs:
       - name: Swap to the production slot
         run: |
           az webapp deployment slot swap --resource-group ${{ vars.RESOURCE_NAME_PREFIX }}-rg --name ${{ vars.WEBAPP_NAME }} --slot ${{ vars.WEBAPP_SLOT_NAME }} --target-slot production
+
+  deploy-to-production:
+    runs-on: ubuntu-22.04
+    needs: [create-and-publish-image, deploy-to-dev]
```
Comment (sam-c-dfe) [inline .github/workflows/main-build-and-deploy.yml:126]: The array needs deploy-to-dev switching out for the staging job

**Review by sam-c-dfe — CHANGES_REQUESTED** [review]

File: .github/workflows/main-build-and-deploy.yml (hunk 2 above)
Comment (sam-c-dfe) [reply .github/workflows/main-build-and-deploy.yml:126]: Just for additional context, if this was left as is, the deployment to staging and production would happen at the same time. Changing this to deploy-to-staging will cause it to wait :-)

File: .github/workflows/main-build-and-deploy.yml (hunk 1 above)
Comment (RobertGHippo) [reply .github/workflows/main-build-and-deploy.yml:127]: Well spotted, fixing now.

File: .github/workflows/main-build-and-deploy.yml (hunk 2 above)
Comment (RobertGHippo) [reply .github/workflows/main-build-and-deploy.yml:126]: Well spotted, fixing now.

**Review by sam-c-dfe — APPROVED** [review]

**Pull request merged**